### PR TITLE
add package: mingw-w64-zxing-cpp

### PR DESCRIPTION
**Pull request by cocomeow**

### CODE REVIEW

File: mingw-w64-zxing-cpp/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+pkgver=1.2.0
+pkgrel=1
+pkgdesc='A C++ library to decode QRCode'
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+url='https://github.com/nu-book/zxing-cpp'
+license=('Apache')
+depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
+             "${MINGW_PACKAGE_PREFIX}-cmake")
```
Comment (MehdiChinoune) [inline mingw-w64-zxing-cpp/PKGBUILD:14]:
```suggestion
             "${MINGW_PACKAGE_PREFIX}-cmake"
             "${MINGW_PACKAGE_PREFIX}-ninja")
```
ninja is preferred.

File: mingw-w64-zxing-cpp/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+build() {
+  # Shared Build
+  [[ -d "build-${MINGW_CHOST}-shared" ]] && rm -rf "build-${MINGW_CHOST}-shared"
+
+  mkdir -p ${srcdir}/build-${MINGW_CHOST}-shared
+  cd ${srcdir}/build-${MINGW_CHOST}-shared
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  ${MINGW_PREFIX}/bin/cmake.exe \
+  -G"MSYS Makefiles" \
```
Comment (MehdiChinoune) [inline mingw-w64-zxing-cpp/PKGBUILD:29]:
```suggestion
  -G"Ninja" \
```

**Comment by MehdiChinoune** [comment]
It is recommended to install static library before the shared one, so cmake install the exported cmake files of the shared library (overwrite the ones from tha static library).

File: mingw-w64-zxing-cpp/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+package() {
+  # Shared Install
+  cd "${srcdir}/build-${MINGW_CHOST}-shared"
+  make DESTDIR=${pkgdir} install
+
+  # Static Install
+  cd "${srcdir}/build-${MINGW_CHOST}-static"
+  make DESTDIR=${pkgdir} install
```
Comment (MehdiChinoune) [inline mingw-w64-zxing-cpp/PKGBUILD:65]:
```suggestion
  # Static Install
  cd "${srcdir}/build-${MINGW_CHOST}-static"
  DESTDIR=${pkgdir} cmake --install .

  # Shared Install
  cd "${srcdir}/build-${MINGW_CHOST}-shared"
  DESTDIR=${pkgdir} cmake --install .
```

**Comment by cocomeow** [comment]
> It is recommended to install static library before the shared one, so cmake install the exported cmake files of the shared library (overwrite the ones from tha static library).

Thank you for your suggestion!

**Review by MehdiChinoune — APPROVED** [review]

**Comment by Biswa96** [comment]
The later "MSYS Makefiles" also need to change Ninja generator. BTW, one single commit will look clean.

File: mingw-w64-zxing-cpp/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  # Static Build
+  [[ -d "build-${MINGW_CHOST}-static" ]] && rm -rf "build-${MINGW_CHOST}-static"
+
+  mkdir -p ${srcdir}/build-${MINGW_CHOST}-static
+  cd ${srcdir}/build-${MINGW_CHOST}-static
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  ${MINGW_PREFIX}/bin/cmake.exe \
+  -G"MSYS Makefiles" \
```
Comment (MehdiChinoune) [inline mingw-w64-zxing-cpp/PKGBUILD:48]:
```suggestion
  -G"Ninja" \
```

**Review by MehdiChinoune — APPROVED** [review]

**Comment by cocomeow** [comment]
> The later "MSYS Makefiles" also need to change Ninja generator. BTW, one single commit will look clean.

Do I need to do the next step by myself, or wait for the maintainer to squash merge pull requests?

**Comment by MehdiChinoune** [comment]
> > The later "MSYS Makefiles" also need to change Ninja generator. BTW, one single commit will look clean.
> 
> Do I need to do the next step by myself, or wait for the maintainer to squash merge pull requests?

Do it yourself, They generally merge them as they are.

**Comment by MehdiChinoune** [comment]
You can squash the commits and then force-push your branch, no need to open a new PR!

**Comment by cocomeow** [comment]
> You can squash the commits and then force-push your branch, no need to open a new PR!

Sorry for my lack experience in this processes, the master branch of fork repository was force-pushed and could not open again, please move to the new pull-request this time.

**Pull request closed without merging**